typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -823,6 +840,15 @@
      [((? DepFun? dfun) _)
       (for/or ([a1 (in-list arrows1)])
         (arrow-subtype-dfun* A a1 dfun))]
+     [((or (FlatCon: t2-pre t2-post)
+           (Con: t2-pre t2-post))
+       _)
+      ;; XXX hacks
+      (match t1
```

Comment:
XXX hacks, oh my! Um... either get rid of the hacks, or document the hacks? I don't know what this means, what the consequences are, etc.